test/integration/ratelimit_integration_test.cc

```diff
@@ -157,10 +162,19 @@ class RatelimitIntegrationTest : public HttpIntegrationTest,
 
   const uint64_t request_size_ = 1024;
   const uint64_t response_size_ = 512;
+  bool failure_mode_ = false;
```

Comment:
the integration test and the unit tests use the same varaible name `failure_mode_` in opposite ways. In the unit tests we use it directly for the `failure_mode_allow` config value, and here we use it logically opposite, i.e if `failure_mode_` is true then `failure_mode_allow` is false. Can we make it consistent between the tests?